### DISCUSSION TITLE
Create `code-line-numbers` across code chunks?

### DISCUSSION THREAD

**Discussion by ChandlerLutz** [quarto-dev/quarto-cli#7756] [Q&A] [answered]
### Description

I'm using `code-line-numbers` to add line numbers to code chunks. Is it possible to number lines across code chunks (e.g., not restart at number 1) with setting `code-line-numbers` manually within each chunk?

**Comment by cderv** [comment]
No I don't think this is possible. Currently this feature relies mainly on Pandoc's highlighter support (https://pandoc.org/MANUAL.html#extension-fenced_code_attributes) and it applies on code block by code block.

**Comment by hturner** [comment]
This is now possible, as documented at the link @cderv provided, e.g.

````
```{.r startFrom="100"}
```
````

would be nice if this was documented in the Quarto docs.

**Reply by cderv** [reply]
I suppose you mean the `startFrom` option ? 

You probably need to wrap with more backtick you code cell you wanted to show.

**Reply by mcanouil** [reply]
_(I edited to add the extra backticks around the code. I do not know since when I can in discussions ...)_

@hturner  If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

**Reply by hturner** [reply]
Well, I was just showing how to specify the option. If you want an example with code, here's one I'm working on, to start from where the server function is defined in an app.R file

````
## Writing a Shiny App {auto-animate="true"}

```{.r filename="app.R" startFrom=31}
)

server <- function(input, output, session) {

}
```

## Writing a Shiny App {auto-animate="true"}

```{.r filename="app.R" startFrom=31}
)

server <- function(input, output, session) {
    # Reactive value to store the uploaded data
    uploaded_data <- reactive({
        req(input$file)
        data <- read.csv(input$file$datapath, header = TRUE)
        return(data)
    })
}
```
````

**Reply by hturner** [reply]
Ah, thanks for making the edit, your comment makes more sense now!

**Reply by cderv** [reply, marked as answer]
We'll document, but we'll do it by a new option that is more aligned with the context, and findable with autocompletion. Current candidate is `code-line-numbers-start`

- https://github.com/quarto-dev/quarto-cli/issues/13042
